src/http/httpClient/Middleware/BodyInspectionHandler.cs

```diff
@@ -0,0 +1,95 @@
+// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Kiota.Http.HttpClientLibrary.Extensions;
+using Microsoft.Kiota.Http.HttpClientLibrary.Middleware.Options;
+
+namespace Microsoft.Kiota.Http.HttpClientLibrary.Middleware;
+
+/// <summary>
+/// The Body Inspection Handler allows the developer to inspect the body of the request and response.
+/// </summary>
+public class BodyInspectionHandler : DelegatingHandler
```

Comment:
Similar to the `HeadersInspectionHandler`, should we register this in the default list so that the option can be used without having to create a new client or modify/update the handler list? 
https://github.com/microsoft/kiota-dotnet/blob/216801b4931b03aac86aa9d656a472bf6379bee3/src/http/httpClient/KiotaClientFactory.cs#L91

Reply:
done